File: src/entity/move/MovePos.java
package entity.move;

import design.utilities.Pair;

/**
 * An interface used to set next entity's coordinates and direction in the dungeon 
 */

public interface MovePos {
	
	/**
	 * @param pos, entity's current coordinates 
	 * @param dir, entity's current direction
	 * @return the possible next entity's coordinates and direction
	 */
	Pair<Integer,Integer> nextPos(Pair<Integer,Integer> pos, Direction dir);

}

File: Fabbri_Lucia/Csharp_LuciaFabbri/MovePos.cs
using System;

namespace dynamicBody.move
{
    /// <summary>
    /// An interface used to set next dynamic body's coordinates and direction in the dungeon
    /// </summary>
    public interface IMovePos
    {
        /// <summary>
        /// Method used to get next possible dynamic body's position
        /// </summary>
        /// <param name="pos"> dynamic body's current coordinates </param>
        /// <param name="dir"> dynamic body's current direction </param>
        /// <returns>  the possible next dynamic body's coordinates and direction </returns>
        ValueTuple<int, int> NextPos(ValueTuple<int, int> pos, Direction dir);
    }
}

